localstack/services/stores.py

```diff
@@ -186,8 +238,12 @@ def __getitem__(self, region_name) -> BaseStoreType:
 
         return super().__getitem__(region_name)
 
-    def reset(self):
-        """Clear all store data."""
+    def reset(self, _reset_universal: bool = False):
```

Comment:
nit: why the `_` prefix in the parameter name?

Reply:
what makes universal attributes special that they require special treatment in the reset method?

Reply:
`_` is to indicate that this is an internal argument not meant for public use. 

There are some assumptions on how this API will work. Calling `RegionBundle.reset()` will only reset data for current account. We don't want other accounts to be affected, which will happen if `_reset_universal` is set. That should only happen when `RegionBundle.reset()` is called by an `AccountRegionBundle` which sets `_reset_universal`.

```
ar_bundle: AccountRegionBundle = sample_stores
ar_bundle.reset()
# This will reset all stores in all accounts and all regions
# Including data marked as `CrossAccountAttribute`, which is stored in the `_universal` dict

r_bundle: RegionBundle = sample_stores['001122334455']
r_bundle.reset()
# This will reset all stores in all regions for `001122334455` account
# Including data marked as `CrossRegionAttribute`, which is stored in the `_global` dict
```

Reply:
Added some addditional inline docs in 971fc112

Reply:
There is an [high level arch doc](https://www.notion.so/localstack/Multi-accounts-and-multi-region-f60b280bebe54e8ba99c5b8c0c97bafe) on this but I'm keeping the implementation-specific docs right here :+1: